Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt

```diff
@@ -1,6 +1,72 @@
 package de.rki.coronawarnapp.familytest.ui.selection
 
+import android.os.Bundle
+import android.view.View
 import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.navArgs
 import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.FragmentTestRegistrationSelectionBinding
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.doNavigate
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBinding
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import javax.inject.Inject
 
-class TestRegistrationSelectionFragment : Fragment(R.layout.fragment_test_registration_selection)
+class TestRegistrationSelectionFragment : Fragment(R.layout.fragment_test_registration_selection), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val navArgs by navArgs<TestRegistrationSelectionFragmentArgs>()
+    private val viewModel: TestRegistrationSelectionViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as TestRegistrationSelectionViewModel.Factory
+            factory.create(navArgs.coronaTestQrCode)
+        }
+    )
+    private val binding: FragmentTestRegistrationSelectionBinding by viewBinding()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.routeToScreen.observe2(this) {
```

Comment:
This screen stays in the navigation back stack even after the test is registered, I think when we navigate from here, it should be piped up from the stack